### PR TITLE
[AAQ-752] Update tables for aam

### PR DESCRIPTION
**Pull request by suzinyou**
Reviewer: @sidravi1 
Estimate: 30 minutes

---

## Ticket

Fixes: https://idinsight.atlassian.net/browse/AAQ-752

### Goal

Update table so that it is more easily query-able (especially for AAM)

### Changes

1. Use `_` in table names
2. Add `user_id` column
3. Add migration script
4. Modify tests and scripts

## Checklist

Fill with `x` for completed. 

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts

(Delete any items below that are not relevant)
- [x] I have updated the automated tests
- [x] I have updated the scripts in `scripts/`

### CODE REVIEW

File: core_backend/migrations/versions/2024_08_01_465368ca2bac_rename_tables_and_add_user_id_column.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+# revision identifiers, used by Alembic.
+revision: str = "465368ca2bac"  # pragma: allowlist secret
+down_revision: Union[str, None] = "9d0519332fdd"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # Rename tables
```
Comment (suzinyou) [inline core_backend/migrations/versions/2024_08_01_465368ca2bac_rename_tables_and_add_user_id_column.py:22]: The DDL test was failing. I looked at the DDL statement of upgraded tables and added the index operations.

File: core_backend/migrations/versions/2024_08_01_465368ca2bac_rename_tables_and_add_user_id_column.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    # )
+    # op.execute(
+    #     """
+    #     UPDATE content_tag ct
+    #     SET user_id = c.user_id
+    #     FROM content c
+    #     WHERE ct.content_id = c.content_id
+    #     """
+    # )
+    # op.alter_column("content_tag", "user_id", nullable=False)
```
Comment (suzinyou) [inline core_backend/migrations/versions/2024_08_01_465368ca2bac_rename_tables_and_add_user_id_column.py:137]: To do: find out how content_tag table is used and decide whether to include user_id column or not.

File: scripts/add_dummy_data_to_db.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # This script is useful if you want to test the dashboard with dummy data.
 # Navigate to the root directory of the project and run the following command:
+#   > make setup-dev
```
Comment (suzinyou) [inline scripts/add_dummy_data_to_db.py:3]: Script fails without adding 1st user to DB because of non-null user_id constraint

**Review by sidravi1 — COMMENTED** [review]
Thanks @suzinyou. Looks good so far. I'll do a final review tomorrow and merge.

File: scripts/add_dummy_data_to_db.py [hunk 4]
```diff
@@ -165,6 +168,7 @@ def create_feedback_record(
     feedback_db = ResponseFeedbackDB(
         feedback_datetime_utc=dt,
         query_id=query_id,
+        user_id=1,
```
Comment (sidravi1) [inline scripts/add_dummy_data_to_db.py:171]: Should we use `user_id` here as well?

File: scripts/add_dummy_data_to_db.py [hunk 5]
```diff
@@ -184,6 +188,7 @@ def create_content_feedback_record(
         feedback_db = ContentFeedbackDB(
             feedback_datetime_utc=dt,
             query_id=query_id,
+            user_id=1,
```
Comment (sidravi1) [inline scripts/add_dummy_data_to_db.py:191]: And here?

File: core_backend/migrations/versions/2024_08_01_465368ca2bac_rename_tables_and_add_user_id_column.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
+"""Rename tables and add user_id column
+
+Revision ID: 465368ca2bac
+Revises: 9d0519332fdd
```
Comment (sidravi1) [inline core_backend/migrations/versions/2024_08_01_465368ca2bac_rename_tables_and_add_user_id_column.py:4]:
```suggestion
Revises: b87b336df479
```

File: core_backend/migrations/versions/2024_08_01_465368ca2bac_rename_tables_and_add_user_id_column.py [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+"""
+
+from typing import Sequence, Union
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision: str = "465368ca2bac"  # pragma: allowlist secret
+down_revision: Union[str, None] = "9d0519332fdd"
```
Comment (sidravi1) [inline core_backend/migrations/versions/2024_08_01_465368ca2bac_rename_tables_and_add_user_id_column.py:16]:
```suggestion
down_revision: Union[str, None] = "b87b336df479"
```

**Review by sidravi1 — APPROVED** [review]
Thanks @suzinyou! Mind making the few minor changes and fixing the migration down revision and merging it in?

**Pull request merged**